File: 進階物件導向專示範/撲克牌 21點 C# 版本/Player/HumanPlayer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PokerCardGame
{
    class HumanPlayer : Player
    {

        protected override string getName()
        {
            Console.WriteLine("Input your name : ");
            return Console.ReadLine().Trim();
        }

        protected override bool chooseReceiveNextCardOrNot()
        {
            Console.WriteLine("Would you like to get the next card ? (y/n)");
            bool result = char.ToUpper(Console.ReadKey().KeyChar) == 'Y';
            Console.WriteLine();
            return result;
        }

    }
}

File: Joanna/卡牌戰鬥遊戲/src/Player/HumanPlayer.java
package Player;

import java.util.Scanner;

public class HumanPlayer extends Player {
	
	@Override
	protected String nameSelf() {
		Scanner input = new Scanner(System.in);
		System.out.print("¿é¤Jª±®a¦WºÙ¡G");
		name = input.next();
		return name;
	}

	@Override
	public int selectCard() {
		Scanner input = new Scanner(System.in);
		int selection = input.nextInt();
		return selection;
	}
	
}
